hack/update-deps.sh

```diff
@@ -23,8 +23,26 @@ set -o pipefail
 
 cd ${ROOT_DIR}
 
+# The list of dependencies that we track at HEAD and periodically
+# float forward in this repository.
+FLOATING_DEPS=(
+  "knative.dev/test-infra"
+)
+
+# Parse flags to determine any we should pass to dep.
+DEP_FLAGS=""
+while [[ $# -ne 0 ]]; do
+  parameter=$1
+  case ${parameter} in
+    --upgrade) DEP_FLAGS="-update ${FLOATING_DEPS[@]}" ;;
```

Comment:
can you try this script with two or more floaters? I thought there should only be one `-update` flag?

Reply:
OH nevermind... I forgot we are passing in --upgrade, not the deps.

IGNORE ME

also...

:fire: bash

Reply:
This only yields one flag, and I did try with two.  I have a tweak to this incoming to fix the error where `""` is passed to `dep` without `--upgrade`, testing locally now.